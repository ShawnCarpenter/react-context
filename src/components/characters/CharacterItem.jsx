
import React from 'react';
import PropTypes from 'prop-types';

const CharacterItem = ({ name, photoUrl }) => {
  return (
    <figure>
      <img src={photoUrl} alt={name} />
      <figcaption>
        {name}
      </figcaption>
    </figure>
  );
};

CharacterItem.propTypes = {
  name:PropTypes.string.isRequired,
  photoUrl:PropTypes.string.isRequired
};

export default CharacterItem;
