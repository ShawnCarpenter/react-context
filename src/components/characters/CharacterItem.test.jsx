import React from 'react';
import { render } from '@testing-library/react';
import CharacterItem from './CharacterItem';

describe('CharacterItem component', () => {
  it('renders CharacterItem', () => {
    const { asFragment } = render(<CharacterItem
      name="Avatar"
      photoUrl="http://photo.com/img.jpg"/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
