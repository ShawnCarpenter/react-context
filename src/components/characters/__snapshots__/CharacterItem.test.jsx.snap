// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterItem component renders CharacterItem 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="Avatar"
      src="http://photo.com/img.jpg"
    />
    <figcaption>
      Avatar
    </figcaption>
  </figure>
</DocumentFragment>
`;
